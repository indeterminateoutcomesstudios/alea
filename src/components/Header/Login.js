// @flow
import React from 'react'
import Button from './Button'

type Props = {
  showLogin: Function
}

const Login = (props: Props) => {
  const { showLogin } = props

  return <Button onClick={showLogin}>Login</Button>
}

export default Login
