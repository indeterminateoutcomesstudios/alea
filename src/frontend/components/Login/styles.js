// @flow
import styled from 'styled-components'
import { fontSize, colors } from 'frontend/styles/common'
import InputBase from 'frontend/components/Input'
import ButtonBase from 'frontend/components/Button'

// @flow
export const Wrapper = styled.div`
  width: 300px;
  margin: auto;
  background: ${props => props.theme.background};
  border-radius: 5px;
  padding: 2rem;
  box-shadow: 0 5px 10px rgba(0, 0, 0, 0.3);
`

export const Heading = styled.h1`
  text-align: center;
  margin: 0;
  margin-bottom: 2rem;
  padding: 1rem;
  line-height: 0;
`

export const Input = InputBase.extend`
  font-size: ${fontSize.normal};
  padding: 2rem 1rem;
  background: none;
  border: 1px solid ${props => props.theme.borderColor};
  width: 100%;
  margin-bottom: 2rem;
`

export const Button = ButtonBase.extend`
  display: block;
  width: 100%;
  margin-bottom: 2rem;
`

export const Error = styled.div`
  margin-bottom: 2rem;
  color: ${colors.red};
  font-style: italic;
`

export const Forgot = styled.div`
  text-align: center;

  & a,
  & a:link,
  & a:visited {
    color: ${colors.lightBlue};
    text-decoration: none;
  }

  & a:hover {
    color: ${colors.blue};
  }
`
