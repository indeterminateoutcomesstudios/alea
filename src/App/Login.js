/* @flow */
import React from 'react'
import styled from 'styled-components'
import { connect } from 'react-redux'
import { compose, withState, withHandlers } from 'recompose'
// import Button from 'App/components/Button'
import { login } from 'actions'
import { header } from 'styles/fonts'

const enhance = compose(
  connect(null, (dispatch: Function) => ({
    login: (email: string, password: string) => dispatch(login(email, password))
  })),
  withState('email', 'setEmail', ''),
  withState('password', 'setPassword', ''),
  withHandlers({
    onEmailChange: ({ setEmail }) => (e: Object) => setEmail(e.target.value),
    onPasswordChange: ({ setPassword }) => (e: Object) =>
      setPassword(e.target.value),
    onLogin: ({ login, email, password, setPassword }) => (e: Object) => {
      e.preventDefault()
      e.stopPropagation()

      login(email, password)
      setPassword('')
    }
  })
)

type Props = {
  email: string,
  password: string,
  onEmailChange: Function,
  onPasswordChange: Function,
  onLogin: Function
}

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: stretch;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
`

const Login = styled.div`
  width: 300px;
  align-self: center;
  margin: auto;
  background: ${props => props.theme.background};
  border-radius: 5px;
  padding: 2.4rem;
  box-shadow: 0 5px 10px rgba(0, 0, 0, 0.3);
`

const Heading = styled.h1`
  font-family: ${header};
  text-align: center;
  margin: 0;
  margin-bottom: 1.2rem;
  line-height: 3.6rem;
`

// FIXME: Extend input element
const Input = styled.input`
  border-radius: 0;
  background: none;
  border: none;
  padding: 0;
  border-bottom: 1px solid ${props => props.theme.borderColor};
  width: 100%;
  margin-bottom: 1.2rem;
`

// FIXME: Extend button element
const Button = styled.button`
  display: block;
  position: relative;
  margin: 0 auto 0;
`

export default enhance((props: Props) => {
  return (
    <Container>
      <Login>
        <Heading>Aleamancer</Heading>
        <form onSubmit={props.onLogin}>
          <Input
            name="username"
            type="text"
            placeholder="email"
            value={props.email}
            onChange={props.onEmailChange}
          />
          <Input
            name="password"
            type="password"
            placeholder="password"
            value={props.password}
            onChange={props.onPasswordChange}
          />
          <Button onClick={() => {}}>Login</Button>
        </form>
      </Login>
    </Container>
  )
})
