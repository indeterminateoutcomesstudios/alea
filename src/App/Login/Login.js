/* @flow */
import React from 'react'
import { connect } from 'react-redux'
import Button from 'App/components/Button'
import { login } from 'actions'
import styles from './style.css'

type Props = {
  login: Function
}

type State = {
  email: string,
  password: string
}

class Login extends React.Component<*, Props, *> {
  state: State

  constructor (props: Props) {
    super(props)
    this.state = { email: '', password: '' }
  }

  handleEmail (event: Object) {
    this.setState({email: event.target.value})
  }

  handlePassword (event: Object) {
    this.setState({password: event.target.value})
  }

  handleSubmit = (event: Object) => {
    console.log('submitting login')
    event.preventDefault()
    event.stopPropagation()

    this.props.login(
      this.state.email,
      this.state.password
    )
    this.setState({email: '', password: ''})
  }

  render () {
    return (
      <div className={ styles.container }>
        <div className={ styles.login }>
          <h1 className={ styles.title }>Aleamancer</h1>
          <form onSubmit={this.handleSubmit}>
            <input
              className={ styles.input }
              type='text'
              placeholder='email'
              value={this.state.email}
              onChange={e => this.handleEmail(e)}
            />
            <input
              className={ styles.input }
              type='password'
              placeholder='password'
              value={this.state.password}
              onChange={e => this.handlePassword(e)}
            />
            <Button onClick={() => {}} customClass={styles.button}>Login</Button>
          </form>
        </div>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch: Function) => ({
  login: (email: string, password: string) => dispatch(login(email, password))
})

export default connect(null, mapDispatchToProps)(Login)
