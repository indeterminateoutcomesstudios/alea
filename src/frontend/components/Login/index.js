// @flow
import * as React from 'react'
import Logo from 'frontend/components/Logo'
import { Link, Redirect } from 'react-router-dom'
import { Wrapper, Heading, Input, Button, Error, Forgot } from './styles'

type Props = {
  performLogin: Function,
  userId: null | string,
  location: {
    state?: {
      from: string
    }
  }
}

type State = {
  error: string | null
}

export default class Login extends React.Component<Props, State> {
  state = {
    error: null
  }

  username = React.createRef()
  password = React.createRef()

  submit = async (e: SyntheticEvent<>) => {
    const { performLogin } = this.props

    e.preventDefault()
    e.stopPropagation()

    this.setState({ error: null })

    if (!this.username.current || !this.password.current) {
      return
    }

    const username = this.username.current.value
    const password = this.password.current.value

    const result = await performLogin(username, password)
    // See if an error occurred
    if (result.code) {
      if (result.code == 'auth/invalid-email') {
        this.setState({ error: 'The email you entered is invalid.' })
      } else if (result.code == 'auth/wrong-password') {
        this.setState({ error: 'The password is incorrect.' })
      } else if (result.code == 'auth/user-not-found') {
        this.setState({ error: 'There is no account for the given email.' })
      }
      return
    }
  }

  render() {
    const { from } = this.props.location.state || { from: '/sessions' }
    const { userId } = this.props
    const { error } = this.state

    if (userId) {
      return <Redirect to={from} />
    }

    return (
      <Wrapper>
        <Heading>
          <Logo height="26px" />
        </Heading>
        <form onSubmit={this.submit}>
          <Input
            name="username"
            type="text"
            placeholder="email"
            innerRef={this.username}
          />
          <Input
            name="password"
            type="password"
            placeholder="password"
            innerRef={this.password}
          />
          {error && <Error>{error}</Error>}
          <Button large onClick={this.submit}>
            Login
          </Button>
          <Forgot>
            <Link to="/forgot_password">Forgot Password?</Link>
          </Forgot>
        </form>
      </Wrapper>
    )
  }
}
