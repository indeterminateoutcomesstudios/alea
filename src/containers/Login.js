/* @flow */
import React from 'react'
import { connect } from 'react-redux'
import { compose, withState, withHandlers } from 'recompose'
import { login } from 'actions'
import Login from 'components/Login'

const enhance = compose(
  connect(null, (dispatch: Function) => ({
    login: (email: string, password: string) => dispatch(login(email, password))
  })),
  withState('email', 'setEmail', ''),
  withState('password', 'setPassword', ''),
  withHandlers({
    onEmailChange: ({ setEmail }) => (e: Object) => setEmail(e.target.value),
    onPasswordChange: ({ setPassword }) => (e: Object) =>
      setPassword(e.target.value),
    onLogin: ({ login, email, password, setPassword }) => (e: Object) => {
      e.preventDefault()
      e.stopPropagation()

      login(email, password)
      setPassword('')
    }
  })
)

export default enhance(Login)
