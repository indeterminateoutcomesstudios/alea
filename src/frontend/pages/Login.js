// @flow
import { connect } from 'react-redux'
import { compose, withProps } from 'recompose'
import performLogin from 'frontend/firebase/login'
import Login from 'frontend/components/Login'
import type { State } from 'frontend/store'

const mapStateToProps = (state: State) => ({
  userId: state.currentUser.id
})

export default compose(
  connect(mapStateToProps, {}),
  withProps({ performLogin })
)(Login)
