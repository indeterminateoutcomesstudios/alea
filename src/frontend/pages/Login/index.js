// @flow
import * as React from 'react'
import { connect } from 'react-redux'
import Header from 'frontend/containers/Header'
import performLogin from 'frontend/firebase/login'
import Login from 'frontend/components/Login'
import { Background } from './styles'
import type { State } from 'frontend/store'

const LoginView = (props: any) => (
  <Background>
    <Header />
    <Login {...props} performLogin={performLogin} />
  </Background>
)

const mapStateToProps = (state: State) => ({
  userId: state.currentUser.id
})

export default connect(mapStateToProps, {})(LoginView)
