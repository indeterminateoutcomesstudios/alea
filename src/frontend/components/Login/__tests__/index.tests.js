// @flow
import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import Login from '../index'

describe('Login component', () => {
  const renderer = new ShallowRenderer()

  it('renders correctly', () => {
    renderer.render(
      <Login location={{}} userId={null} performLogin={() => {}} />
    )
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot()
  })
})
