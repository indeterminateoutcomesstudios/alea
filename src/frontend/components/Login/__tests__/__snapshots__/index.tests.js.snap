// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders correctly 1`] = `
<styled.div>
  <styled.h1>
    <Logo
      height="26px"
    />
  </styled.h1>
  <form
    onSubmit={[Function]}
  >
    <styled.input
      innerRef={
        Object {
          "current": null,
        }
      }
      name="username"
      placeholder="email"
      type="text"
    />
    <styled.input
      innerRef={
        Object {
          "current": null,
        }
      }
      name="password"
      placeholder="password"
      type="password"
    />
    <styled.button
      large={true}
      onClick={[Function]}
    >
      Login
    </styled.button>
    <styled.div>
      <Link
        replace={false}
        to="/forgot_password"
      >
        Forgot Password?
      </Link>
    </styled.div>
  </form>
</styled.div>
`;
